core/job/spec.go

```diff
@@ -162,12 +165,39 @@ func (s Specs) ToFullNameAndSpecMap(projectName tenant.ProjectName) map[FullName
 	return fullnameAndSpecMap
 }
 
+func (s Specs) Validate() ([]*Spec, error) {
+	me := errors.NewMultiError("validate specs errors")
+	isJobNameValid := map[Name]bool{}
+
+	// validate duplication
+	for _, spec := range s {
+		if valid, ok := isJobNameValid[spec.Name()]; !ok {
+			isJobNameValid[spec.Name()] = true
+		} else if valid { // duplication detected
+			isJobNameValid[spec.Name()] = false // make it invalid
+			me.Append(fmt.Errorf("duplicate %s", spec.Name()))
+		}
+	}
+
+	validSpecs := []*Spec{}
+	for _, spec := range s {
+		if isJobNameValid[spec.Name()] {
+			validSpecs = append(validSpecs, spec)
+		}
+	}
+
+	return validSpecs, me.ToErr()
+}
+
 type Name string
 
 func NameFrom(name string) (Name, error) {
 	if name == "" {
 		return "", errors.InvalidArgument(EntityJob, "name is empty")
 	}
+	if len(name) > maxJobNameLength {
```

Comment:
This needs to be confirmed, because if there's one or more jobs in database which have name length more than specified, this can cause error. If the assumption is there is one or more people outside which uses Optimus, then we can put a script which update the name. But, if the assumption is no other people using Optimus, then we can have the database modification somewhere else.

Reply:
confirmed that we don't have any job name more than 125 chars long, this should be fine then